portal-impl/src/com/liferay/portal/service/impl/ResourcePermissionLocalServiceImpl.java

```diff
@@ -362,8 +362,11 @@ public ResourcePermission fetchResourcePermission(
 			resourcePermissions.size());
 
 		for (ResourcePermission resourcePermission : resourcePermissions) {
+			Set<String> emptySet = Collections.emptySet();
+
 			if (resourcePermission.getActionIds() == 0) {
-				continue;
+				roleIdsToActionIds.put(
```

Comment:
You probably imported the wrong Collections class, like the one that comes
first in the IDE is often from backport-concurrent which is not generic.